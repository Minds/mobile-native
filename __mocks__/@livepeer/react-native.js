// mock @livepeer/react-native
export const LivepeerConfig = {
  setConfig: jest.fn(),
  getConfig: jest.fn(),
  get: jest.fn(),
  set: jest.fn(),
};

export const createReactClient = jest.fn();
export const createLivepeerClient = jest.fn();
export const Player = jest.fn();
