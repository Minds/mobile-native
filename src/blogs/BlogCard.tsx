//@ts-nocheck
import React, { PureComponent } from 'react';

import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';

import { Avatar } from 'react-native-elements';

import FastImage from 'react-native-fast-image';
import SmartImage from '../common/components/SmartImage';
import formatDate from '../common/helpers/date';
import { FLAG_VIEW } from '../common/Permissions';
import Actions from '../newsfeed/activity/Actions';
import { CommonStyle as CS } from '../styles/Common';
import ThemedStyles from '../styles/ThemedStyles';
import type BlogModel from './BlogModel';
import BlogActionSheet from './BlogActionSheet';

type PropsType = {
  entity: BlogModel;
  navigation: any;
};

/**
 * Blog Card
 */
export default class BlogCard extends PureComponent<PropsType> {
  /**
   * Navigate to blog
   */
  navToBlog = () => {
    if (!this.props.navigation || !this.props.entity.can(FLAG_VIEW, true)) {
      return;
    }
    return this.props.navigation.push('BlogView', { blog: this.props.entity });
  };

  /**
   * Trim and remove new line char
   * @param {string} title
   */
  cleanTitle(title) {
    if (!title) {
      return '';
    }
    return title.trim().replace(/\n/gm, ' ');
  }

  /**
   * Render Card
   */
  render() {
    const blog = this.props.entity;
    const channel = this.props.entity.ownerObj;
    const image = blog.getBannerSource();
    const title = this.cleanTitle(blog.title);
    const theme = ThemedStyles.style;

    return (
      <View>
        <View style={[styles.actionSheet]}>
          <BlogActionSheet entity={blog} navigation={this.props.navigation} />
        </View>
        <TouchableOpacity
          onPress={this.navToBlog}
          style={theme.backgroundSecondary}>
          <FastImage
            source={image}
            style={styles.banner}
            resizeMode={FastImage.resizeMode.cover}
          />
          <View style={[CS.padding2x]}>
            <View style={[CS.columnAlignStart, CS.fullWidth]}>
              <Text
                style={[CS.fontL, CS.fontMedium, CS.flexContainer]}
                numberOfLines={2}
                ellipsizeMode="tail">
                {title}
              </Text>
              <View
                style={[
                  CS.marginBottom2x,
                  CS.marginTop2x,
                  CS.rowJustifyCenter,
                  CS.alignCenter,
                ]}>
                {channel && (
                  <Avatar
                    width={24}
                    height={24}
                    rounded
                    source={channel.getAvatarSource()}
                  />
                )}
                <Text
                  style={[CS.fontS, CS.paddingLeft, CS.flexContainer]}
                  numberOfLines={1}>
                  {blog.ownerObj && blog.ownerObj.username.toUpperCase()}
                </Text>
                <Text style={[CS.fontXS, CS.paddingLeft]}>
                  {formatDate(blog.time_created)}
                </Text>
              </View>
            </View>
          </View>
          {!this.props.hideTabs && (
            <Actions entity={blog} navigation={this.props.navigation} />
          )}
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  banner: {
    alignItems: 'stretch',
    flexDirection: 'row',
    height: 150,
    width: '100%',
  },
  actionSheet: {
    position: 'absolute',
    top: 0,
    right: 0,
    zIndex: 1,
  },
});
