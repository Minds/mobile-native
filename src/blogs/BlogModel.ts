import { ImageURISource } from 'react-native';
import { decorate, observable } from 'mobx';
import { MINDS_ASSETS_CDN_URI } from '../config/Config';
import api from '../common/services/api.service';
import { LICENSES } from '../common/services/list-options.service';
import ActivityModel from '../newsfeed/ActivityModel';

/**
 * User model
 */
export default class BlogModel extends ActivityModel {
  license: string = '';
  slug: string = '';

  /**
   * Get banner source
   */
  getBannerSource(): ImageURISource {
    const uri =
      this.thumbnail_src &&
      this.thumbnail_src !== 'https://cdn.minds.com/thumbProxy?src=&c=2708'
        ? this.thumbnail_src
        : `${MINDS_ASSETS_CDN_URI}front/dist/assets/logos/placeholder-bulb.jpg`;

    return { uri, headers: api.buildHeaders() };
  }

  /**
   * Get the license text
   */
  getLicenseText(): string {
    const lic = LICENSES.find(license => this.license === license.value);
    if (!lic) {
      return '';
    }
    return lic.text;
  }
}

/**
 * Define model observables
 */
//@ts-ignore
decorate(BlogModel, {
  description: observable,
});
