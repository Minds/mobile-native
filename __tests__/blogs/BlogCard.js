import React from 'react';
import { Platform, Linking } from 'react-native';
import { render, fireEvent } from '@testing-library/react-native';
import BlogCard from '~/blogs/BlogCard';
import blogFakeFactory from '../../__mocks__/fake/blogs/BlogFactory';
import BlogModel from '~/blogs/BlogModel';
import sp from '~/services/serviceProvider';

jest.mock('~/services/serviceProvider');

// mock services
sp.mockService('styles');
sp.mockService('api');
sp.mockService('i18n');

Linking.openURL = jest.fn();

jest.mock('~/newsfeed/activity/Actions', () => 'Actions');

describe('blog card component', () => {
  it('should render correctly', () => {
    const blogEntity = BlogModel.create(blogFakeFactory(1));

    const { toJSON } = render(<BlogCard entity={blogEntity} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should nav to blog', () => {
    const blogEntity = BlogModel.create(blogFakeFactory(1));
    blogEntity.can = () => true;

    const navigation = { push: jest.fn() };

    const { getByTestId } = render(
      <BlogCard entity={blogEntity} navigation={navigation} />,
    );

    Platform.OS = 'ios';

    // Assuming there's a touchable element with testID="blogCard"
    fireEvent.press(getByTestId('blogPressable'));

    expect(navigation.push).toHaveBeenCalledWith('BlogView', {
      blog: blogEntity,
    });
  });
});
