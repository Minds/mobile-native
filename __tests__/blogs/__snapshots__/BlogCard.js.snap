// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blog card component should renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#1A2025",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": undefined,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "alignItems": "stretch",
            "flexDirection": "row",
            "height": 150,
            "width": "100%",
          },
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "headers": Object {
              "App-Version": "3.8.0",
              "Cache-Control": "no-cache, no-store, must-revalidate",
              "Content-Type": "application/json",
              "Pragma": "no-cache",
              "no-cache": "1",
            },
            "uri": "https://www.minds.com/fs/v1/banners/836607056186159104/1524838665",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "padding": 8,
        }
      }
    >
      <View
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
              },
              Object {
                "fontFamily": "Roboto",
              },
              Array [
                Object {
                  "fontSize": 18,
                },
                Object {
                  "fontFamily": "Roboto-Medium",
                  "fontWeight": "500",
                },
                Object {
                  "flex": 1,
                },
              ],
            ]
          }
        >
          Some Title
        </Text>
        <View
          style={
            Array [
              Object {
                "marginBottom": 8,
              },
              Object {
                "marginTop": 12,
              },
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "alignItems": "center",
              },
            ]
          }
        >
          <View
            height={26}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderRadius": 17,
                "height": 34,
                "width": 34,
              }
            }
            theme={
              Object {
                "colors": Object {
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
              }
            }
            width={26}
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 17,
                  "flex": 1,
                  "overflow": "hidden",
                  "position": "relative",
                }
              }
            >
              <Image
                borderRadius={17}
                onLoad={[Function]}
                style={
                  Object {
                    "bottom": 0,
                    "flex": 1,
                    "height": null,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": null,
                  }
                }
                testID="RNE__Image"
                theme={
                  Object {
                    "colors": Object {
                      "black": "#242424",
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "grey": "rgba(0, 0, 0, 0.54)",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "grey": "#7d7d7d",
                          "primary": "#007aff",
                          "searchBg": "#dcdce1",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                      "white": "#ffffff",
                    },
                  }
                }
                transition={true}
              />
              <View
                accessibilityElementsHidden={false}
                collapsable={false}
                importantForAccessibility="yes"
                nativeID="animatedComponent"
                pointerEvents="auto"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "flex": 1,
                      "height": null,
                      "justifyContent": "center",
                      "width": null,
                    }
                  }
                  testID="RNE__Image__placeholder"
                />
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                    "height": null,
                    "width": null,
                  }
                }
              />
            </View>
          </View>
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                },
                Object {
                  "fontFamily": "Roboto",
                },
                Array [
                  Object {
                    "fontSize": 16,
                  },
                  Object {
                    "paddingLeft": 8,
                  },
                  Object {
                    "flex": 1,
                  },
                ],
              ]
            }
          >
            someusermane
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                },
                Object {
                  "fontFamily": "Roboto",
                },
                Array [
                  Object {
                    "fontSize": 10,
                  },
                  Object {
                    "paddingLeft": 4,
                  },
                ],
              ]
            }
          >
            Apr 27 2018 Â· 07:17
          </Text>
          <View
            style={
              Object {
                "paddingLeft": 4,
              }
            }
          >
            <View>
              <Text
                allowFontScaling={false}
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 28,
                    },
                    Object {
                      "color": "#797B82",
                    },
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                î
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <Actions
      entity={
        Object {
          "access_id": "2",
          "allow_comments": true,
          "boost_rejection_reason": -1,
          "categories": Array [],
          "category": "education",
          "comments:count": undefined,
          "container_guid": "100",
          "custom_meta": Object {
            "author": "",
            "description": "",
            "title": "",
          },
          "description": "Some Blog",
          "draft_access_id": "0",
          "edited": "0",
          "entity_guid": "",
          "excerpt": "Some description",
          "featured": false,
          "featured_id": false,
          "getLockType": [Function],
          "guid": 1,
          "header_bg": "1",
          "header_top": "0",
          "impressions": 100,
          "isOwner": [Function],
          "is_visible": false,
          "last_save": "1524843907",
          "last_updated": "1524838665",
          "license": "attribution-noncommercial-cc",
          "mature": false,
          "mature_visibility": false,
          "message": "",
          "monetized": false,
          "ownerObj": Object {
            "__list": null,
            "access_id": "2",
            "allow_comments": true,
            "banned": "no",
            "blocked": false,
            "boostProPlus": false,
            "boost_autorotate": true,
            "boost_rating": "2",
            "briefdescription": "",
            "canary": false,
            "categories": Array [
              "music",
              "nature",
              "news",
              "politics",
              "science",
              "technology",
              "education",
              "art",
              "health",
            ],
            "chat": true,
            "city": "",
            "confirmEmail": [Function],
            "container_guid": "0",
            "deleted": "0",
            "disable_autoplay_videos": undefined,
            "disabled_boost": false,
            "dob": "",
            "email_confirmed": false,
            "eth_wallet": "0x6fffffdadae350ddd5a1777d5a1777a4f39f0317",
            "fb": false,
            "feature_flags": false,
            "featured_id": "725165605782560768",
            "founder": false,
            "gender": "private",
            "getAvatarSource": [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            },
            "groupsCount": 0,
            "guid": 100,
            "icontime": "1523515420",
            "impressions": 100,
            "isOwner": [Function],
            "is_admin": false,
            "language": "en",
            "last_accepted_tos": 0,
            "legacy_guid": false,
            "mature": "1",
            "mature_channel": "0",
            "mature_visibility": false,
            "merchant": Object {
              "exclusive": Object {
                "amount": "10",
                "enabled": true,
              },
              "id": "acct_Dsdfjdlsaf34243",
              "service": "stripe",
            },
            "mode": 0,
            "monetized": "",
            "name": "Some User",
            "nsfw": Array [],
            "onchain_booster": 0,
            "owner_guid": "0",
            "pending_subscribe": false,
            "permissions": Object {},
            "pinned_posts": Array [],
            "plus": true,
            "plus_method": "tokens",
            "pro": false,
            "pro_method": "tokens",
            "programs": Array [
              "affiliate",
            ],
            "rating": "1",
            "rewards": true,
            "signup_method": false,
            "site_guid": false,
            "social_profiles": Array [],
            "spam": "0",
            "subscribed": false,
            "subscriber": true,
            "subscribers_count": 50,
            "subscriptions_count": 0,
            "subtype": false,
            "tags": Array [],
            "time_created": "1468113204",
            "time_updated": false,
            "type": "user",
            "urn": "",
            "username": "someusermane",
            "verified": true,
            "website": "",
            "wire_rewards": Object {
              "description": "",
              "rewards": Object {
                "money": Array [],
                "points": Array [],
              },
            },
          },
          "owner_guid": "1",
          "paywall": false,
          "paywall_unlocked": false,
          "perma_url": "https://www.minds.com/blog/view/836607056186159104",
          "permissions": Object {},
          "pinned": false,
          "published": true,
          "quotes": 0,
          "rating": 1,
          "reminds": 58,
          "route": "blog/view/836607056186159104",
          "slug": "",
          "subtype": "blog",
          "thumbnail_src": "https://www.minds.com/fs/v1/banners/836607056186159104/1524838665",
          "thumbs:down:count": 1,
          "thumbs:down:user_guids": Array [
            "3",
          ],
          "thumbs:up:count": 2,
          "thumbs:up:user_guids": Array [
            "1",
            "2",
          ],
          "time_created": "1524838665",
          "time_published": "1524838665",
          "time_updated": "1524862542",
          "title": "Some Title",
          "type": "object",
          "urn": "",
          "username": "",
          "wire_threshold": "",
        }
      }
    />
  </View>
</View>
`;
